ast/index.go

```diff
@@ -13,11 +13,8 @@ type Index struct {
 }
 
 func (n *Index) Accept(v Visitor) Node {
-	// the Key may have further interpolations
-	switch n.Key.(type) {
-	case *Call, *VariableAccess:
```

Comment:
Ah I see - by changing to use `key := stack.Pop()` eval phase can now get access to the result of the call node evaluation.